src/modules/hide_prime_promotions/index.js

```diff
@@ -0,0 +1,45 @@
+const $ = require('jquery');
+const settings = require('../../settings');
+const watcher = require('../../watcher');
+
+const TWITCH_PRIME_BANNER_SELECTOR =
+    '[data-test-selector="test_selector_prime_tracking_button_wrapper"] [data-a-target="blue-bar"]';
+
+class HidePrimePromotionsModule {
+    constructor() {
+        settings.add({
+            id: 'hideTwitchPrimeBanner',
+            name: 'Hide Twitch Prime Banner',
+            defaultValue: true,
+            description: 'Hides the Twitch Prime Banner on the homepage'
+        });
+        settings.add({
+            id: 'hidePrimePromotion',
+            name: 'Hide Prime Promotions',
+            defaultValue: false,
+            description: 'Hides the "Free With Prime" section of the sidebar'
+        });
+
+        settings.on('changed.hideTwitchPrimeBanner', () => this.toggleTwitchPrimeBar());
+        settings.on('changed.hidePrimePromotion', () => this.togglePrimePromotions());
+
+        watcher.on('load.homepage', () => this.toggleTwitchPrimeBar());
+        watcher.on('load', () => this.togglePrimePromotions());
+    }
+
+    toggleTwitchPrimeBar() {
+        $(TWITCH_PRIME_BANNER_SELECTOR).toggleClass(
```

Comment:
we prefer to set classes on body instead of targeting elements directly. due to the nature of react elements can be remounted at any moment, triggering custom classes to be purged from the dom